ee/modules/600-flant-integration/hooks/madison/revoke.go

```diff
@@ -84,9 +94,27 @@ func revokeHandler(input *go_hook.HookInput, dc dependency.Container) error {
 		return nil // dont this we need an error
 	}
 
+	// Create a ConfigMap to indicate revocation and prevent re-registration.
 	if madisonResp.Error == "Archived setup" {
-		input.ConfigValues.Remove("flantIntegration.licenseKey")
-		input.ConfigValues.Remove("flantIntegration.madisonAuthKey")
+		// Create CM to indicate revoked license.
+		cm := &v1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ConfigMap",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      revokedCMName,
+				Namespace: revokedCMNamespace,
+				Labels: map[string]string{
+					"heritage": "flant-integration",
+				},
+			},
+		}
+		input.PatchCollector.Create(cm, object_patch.IgnoreIfExists())
+
+		// Remove internal values.
+		input.Values.Remove(internalMadisonKeyPath)
+		input.Values.Remove(internalLicenseKeyPath)
```

Comment:
Seems like a problem. If Prometheus is archived in Madison, we will stop sending metrics and logs.

Reply:
Comment is added about telemetry.